### ISSUE TITLE
Cannot mix long syntax and short syntax volume declaration

### ISSUE THREAD

**Issue by cyqsimon** [stephanbuys/docker-compose-types#31]
## TLDR

If you declare some volumes using the short syntax and some others using the long syntax, the ones using the long syntax will fail parsing. I believe this to be an upstream problem in `serde_yaml`.

## Steps to reproduce

Using [Frigate](https://docs.frigate.video/frigate/installation/) as an example:

```yaml
version: "3.9"
services:
  frigate:
    image: ghcr.io/blakeblackshear/frigate:stable
    volumes:
      - type: tmpfs
        target: /tmp/cache
      # - type: bind
      #   source: /etc/frigate/config.yml
      #   target: /config/config.yml
      - /etc/frigate/config.yml:/config/config.yml
```

```rust
let yaml = include_str!("frigate.yaml");
let parsed: Compose = serde_yaml::from_str(yaml).unwrap();
```

```
Error("services.frigate: data did not match any variant of untagged enum Volumes", line: 4, column: 5)
```

## What I tried

1. Only use short syntax in `volumes`, one or many items - works.
2. Only use long syntax in `volumes`, one or many items - works.
3. Mix long and short syntax - fails.

You can verify by commenting and/or uncommenting lines 8-11 in the aforementioned example.

The reason I suspect this is a `serde_yaml` problem is that the error reminds me of a similar problem I ran into last year: dtolnay/serde-yaml#344. In general `serde_yaml` seems to be ***very*** buggy since the introduction of YAML `!tag` support in 0.9. I've had to downgrade to 0.8 for a few of my own projects.

## Version

`serde_yaml`: 0.9.27
`docker-compose-types`: 0.5.1 & 0.6.0

**Comment by stephanbuys** [comment]
@cyqsimon, thanks for reporting this. It was just an implementation problem.

**Comment by cyqsimon** [comment]
Ah yes. Implementation problem indeed. What was I thinking.

I really shouldn't be coding today 😅. My brain's not working quite right.
